### DISCUSSION TITLE
自定义命令的handler函数触发时，如何获取所选择的单元格或者行信息

### DISCUSSION THREAD

**Discussion by 13701138894** [dream-num/univer#3646] [Q&A | 常见问题] [answered]
在handler中，如何获取右键选择的行信息或者单元格信息
export const reportButtonOperation: ICommand = {
  id: 'report',
  type: CommandType.OPERATION,
  handler: async (accessor: IAccessor) => {
    // console.log(accessor.get(DocSelectionManagerService))
    // console.log(accessor.get(DocSelectionManagerService).getDocRanges())
    // console.log(accessor.get(DocSelectionManagerService).getActiveTextRange())
    // console.log(accessor.get(DocSelectionManagerService).getTextRanges())
    // console.log(accessor.get(DocSelectionManagerService).getSelectionInfo())
    return true
  }
}

**Comment by 13701138894** [comment]
![image](https://github.com/user-attachments/assets/0d4774a1-a4c3-462b-a004-b6ec5785ed8c)

**Comment by hexf00** [comment, marked as answer]
The parameters of the command are determined by the trigger, but you can still get any service in the DI system through inject

Read the following documents to understand the selection content

[univer.ai/guides/sheet/features/core/range-selection#get-active-selection](https://univer.ai/guides/sheet/features/core/range-selection#get-active-selection)

[univer.ai/typedoc/@univerjs/facade/classes/FSelection#getactiverange](https://univer.ai/typedoc/@univerjs/facade/classes/FSelection#getactiverange)

**Reply by JokerLHF** [reply]
has code example？
